llvm/test/CodeGen/AMDGPU/release-vgprs-dbg-loc.mir

```diff
@@ -0,0 +1,40 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+# RUN: llc -mtriple=amdgcn -mcpu=gfx1100 -run-pass=si-insert-waitcnts -o - %s | FileCheck %s
+
+# Verify that si-insert-waitcnts copies debug locations from the s_endpgm to the
+# "dealloc vgprs" s_sendmsg. If these are not present, the debugger will be
+# unable to break at the end of the shader.
```

Comment:
Could you add something like this description to the commit message when you merge? I guess the problem is that when you set a source breakpoint at the end of the program, you want the debugger to put it before the sendmsg not after it, so the vgprs are still available??

Reply:
Sure, I'll add more context to the commit. I just asked @lancesix about this, its not just that the VGPRs are unavailable, the problem is also that deallocating VGPRs makes it impossible to do a context save, which GDB needs to report a breakpoint. See: https://github.com/ROCm/ROCdbgapi/blob/7e22c4d4a8852e1950aea7f9d3e49899e4929fb6/include/amd-dbgapi.h.in#L358-L361